go/vt/vttablet/tabletmanager/vreplication/vreplicator.go

```diff
@@ -224,22 +224,24 @@ func (vr *vreplicator) replicate(ctx context.Context) error {
 	}
 }
 
-// PrimaryKeyInfo is used to store charset and collation for primary keys where applicable
-type PrimaryKeyInfo struct {
-	Name       string
-	CharSet    string
-	Collation  string
-	DataType   string
-	ColumnType string
+// ColumnInfo is used to store charset and collation for primary keys where applicable
+type ColumnInfo struct {
+	Name        string
+	CharSet     string
+	Collation   string
+	DataType    string
+	ColumnType  string
+	IsPK        bool
+	IsGenerated bool
```

Comment:
this becomes very parallel to https://github.com/vitessio/vitess/blob/50365bb24460e919e9875a5fd204e3714a94066d/go/vt/vttablet/onlineddl/vrepl/types.go#L35-L58. I see we are solving the same kind of problems in different places. I'll defer to yours anyway.